### PR TITLE
Extend CarbonBlackEnterpriseEDR to support new API v7 methods. 

### PR DESCRIPTION
**Pull request by Enigmatyk**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Extending CarbonBlackEnterpriseEDR to support API v7 new functionalities.

- Add tags to threats
- Add threat notes
- Add alerts notes
- Get threat tags

PR related to https://github.com/demisto/content/pull/28232 that was closed to get clearance from customer. 

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml [hunk 1]
```diff
@@ -1301,10 +1304,64 @@ script:
...
       type: Number
-  dockerimage: demisto/python3:3.10.13.72123
+  - arguments:
+    - description: Threat ID to add the tags to.
+      isArray: true
+      name: threat_id
+      required: true
+    - description: The tags to be added to the threat.
+      name: tags
+      required: true
```
Comment (YuvHayun) [inline Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml:1314]: Shouldn't this be a list as well?
If so let's add the "Comma-separated list of..." to the description and the isArray: true.

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml [hunk 2]
```diff
@@ -1301,10 +1304,64 @@ script:
...
+    outputs:
+    - contextPath: CarbonBlackEEDR.Threat.ThreatID
+      description: Threat ID.
+    - contextPath: CarbonBlackEEDR.Threat.Tags
+      description: Threat ID tags.
+  - arguments:
+    - description: Threat ID.
+      name: threat_id
+      required: true
+    - description: Notes to be added to the provided threat ID.
```
Comment (YuvHayun) [inline Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml:1326]: Shouldn't this be a list as well?
If so let's add the "Comma-separated list of..." to the description and the isArray: true.

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml [hunk 3]
```diff
@@ -1301,10 +1304,64 @@ script:
...
+    outputs:
+    - contextPath: CarbonBlackEEDR.Threat.ThreatID
+      description: Threat ID.
+    - contextPath: CarbonBlackEEDR.Threat.Notes
+      description: Threat ID notes.
+  - arguments:
+    - description: Alert ID.
+      name: alert_id
+      required: true
+    - description: Notes to be added to the provided alert ID.
```
Comment (YuvHayun) [inline Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml:1340]: Shouldn't this be a list as well?
If so let's add the "Comma-separated list of..." to the description and the isArray: true.

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml [hunk 4]
```diff
@@ -1301,10 +1304,64 @@ script:
...
+      name: notes
+      required: true
+    description: Update alert ID notes.
+    name: cb-eedr-add-alert-notes
+    outputs:
+    - contextPath: CarbonBlackEEDR.Alert.AlertID
+      description: Alert ID to add the notes to.
```
Comment (YuvHayun) [inline Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml:1347]:
```suggestion
      description: The alert ID.
```

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml [hunk 5]
```diff
@@ -1301,10 +1304,64 @@ script:
...
+      required: true
+    description: Update threat ID notes.
+    name: cb-eedr-add-threat-notes
+    outputs:
+    - contextPath: CarbonBlackEEDR.Threat.ThreatID
+      description: Threat ID.
+    - contextPath: CarbonBlackEEDR.Threat.Notes
+      description: Threat ID notes.
+  - arguments:
+    - description: Alert ID.
```
Comment (YuvHayun) [inline Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml:1337]:
```suggestion
    - description: Alert ID to add the notes to.
```

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml [hunk 6]
```diff
@@ -1301,10 +1304,64 @@ script:
...
+      required: true
+    description: Update alert ID notes.
+    name: cb-eedr-add-alert-notes
+    outputs:
+    - contextPath: CarbonBlackEEDR.Alert.AlertID
+      description: Alert ID to add the notes to.
+    - contextPath: CarbonBlackEEDR.Alert.Notes
+      description: Alert notes.
+  - arguments:
+    - description: Provide threat ID.
```
Comment (YuvHayun) [inline Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml:1351]:
```suggestion
    - description: The threat ID for which we wish to get the tags.
```

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR_test.py [hunk 7]
```diff
@@ -171,3 +171,32 @@ def test_event_by_process_failing(mocker, requests_mock, demisto_args, expected_
...
+# Mock response from the API for testing
+MOCK_THREAT_TAGS_RESPONSE = {
+    'list': [
+        {'tag': 'malware'},
+        {'tag': 'suspicious'}
+    ]
+}
+
+
+def test_get_threat_tags(mocker):
```
Comment (YuvHayun) [inline Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR_test.py:185]: Please add tests for the rest of the commands you've added.
Also, for each test, please add a given, when, then documentation (can find examples from other tests in this file.)

Comment (YuvHayun) [inline Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR_test.py:185]: Also, let's test the whole command (get_threat_tags_command) and not just the request part.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hey @Enigmatyk, thx a lot for your contribution, it looks great!
I left some minor comments for you, let me know when you're done and ready for re-review (don't forget to update the readme accordingly).
I'm available both here and via Slack's DFIR channel (Yuval Hayun) if any question comes up.

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml [hunk 8]
```diff
@@ -1,7 +1,4 @@
 category: Endpoint
-sectionOrder:
```
[the remark this answers is not in the document]
Comment (Enigmatyk) [reply Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml:2]: It was from the format command.

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml (hunk 2 above)
Comment (Enigmatyk) [reply Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml:1326]: It's a string.

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml (hunk 3 above)
Comment (Enigmatyk) [reply Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml:1340]: It's a string.

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml (hunk 1 above)
Comment (Enigmatyk) [reply Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.yml]: Fixed, threat is string, tags is list.

File: Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR_test.py (hunk 7 above)
Comment (Enigmatyk) [reply Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR_test.py:287]: tests added.

Comment (Enigmatyk) [reply Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR_test.py:287]: tests added.

**Review by YuvHayun — COMMENTED** [review]
Hey @Enigmatyk, thx for addressing the fixes so fast and very high level.
I've sent you a msg on Slack to address a few more minor changes + to schedule a quick demo.

**Comment by Enigmatyk** [comment]
@YuvHayun changes on tests done, please review it.

**Comment by Enigmatyk** [comment]
Record added.
[Recording-20240116_112038.webm](https://github.com/demisto/content/assets/5115163/b42a6c0e-6fc7-4a05-b004-e8172ddcad9d)

**Review by YuvHayun — APPROVED** [review]

**Comment by YuvHayun** [comment]
Merging to internal PR, thx a lot for your contribution! @Enigmatyk

**Pull request merged**